arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -135,84 +126,106 @@ public inline fun <Error, A> iorNel(noinline combineError: (NonEmptyList<Error>,
   )
 }
 
+/**
+ * Runs a computation [block] using [Raise], and ignore its outcome.
+ *
+ * This function re-throws any exceptions thrown within the [Raise] block.
+ *
+ * Read more about running a [Raise] computation in the
+ * [Arrow docs](https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/#running-and-inspecting-results).
+ */
+public inline fun impure(block: UnitRaise.() -> Unit) {
+  contract { callsInPlace(block, InvocationKind.AT_MOST_ONCE) }
+  return merge { ignoreErrors(Unit, block) }
+}
+
 /**
  * Implementation of [Raise] used by `ignoreErrors`.
  * You should never use this directly.
  */
 public class IgnoreErrorsRaise<N>(
   private val raise: Raise<N>,
-  private val error: () -> N
-) : Raise<Any?> {
+  private val error: N
+) : SingletonRaise<Any?>() {
   @RaiseDSL
-  override fun raise(r: Any?): Nothing =
-    raise.raise(error())
+  override fun raise(): Nothing =
+    raise.raise(error)
+}
+
+public sealed class SingletonRaise<in Error> : Raise<Error> {
```

Comment:
Is there a specific reason for including the `Error` type parameter here? I was thinking that maybe it's possible to make `SingletonRaise : Raise<Any?>`

Reply:
I have the `Error` to restrict what types `raise(foo)` is allowed to accept. That's to match the current behaviour of `nullable`, `option` etc where they only allow raising `null` and `None` respectively. If `SingletonRaise` is `Raise<Any?> by default, that can result in unintentionally ignoring errors, and will likely be a significant breaking change. I will experiment however with perhaps some form of intersection-type shenanigans and will report back.